ddpui/ddpairbyte/airbytehelpers.py

```diff
@@ -1042,12 +1137,62 @@ def get_schema_changes(org: Org):
     """
     Get the schema changes of a connection in an org.
     """
-    org_schema_change = OrgSchemaChange.objects.filter(org=org)
+    org_schema_change = OrgSchemaChange.objects.filter(org=org).select_related("schedule_job").all()
 
     if org_schema_change is None:
         return None, "No schema change found"
 
-    schema_changes = list(org_schema_change.values())
+    large_connections = (
+        ConnectionMeta.objects.filter(
+            connection_id__in=[change.connection_id for change in org_schema_change],
+            schedule_large_jobs=True,
+        )
+        .all()
+        .values_list("connection_id", flat=True)
+    )
+
+    schema_changes = []
+    for change in org_schema_change:
+        schema_changes.append(
+            {
+                **model_to_dict(change, exclude=["org", "id"]),
+                **{
+                    "schedule_job": (
+                        {
+                            "scheduled_at": change.schedule_job.scheduled_at,
+                            "flow_run_id": change.schedule_job.flow_run_id,
+                            "job_type": change.schedule_job.job_type,
+                            "status": None,
+                            "state_name": None,
+                        }
+                        if change.schedule_job
+                        else None
+                    ),
+                    "is_connection_large": change.connection_id in large_connections,
+                    "next_job_at": get_schedule_time_for_large_jobs(),
```

Comment:
can we put `if change.schedule_job else None` here as well? just so that in future we don't get confused when looking directly at the database

Reply:
When do you do the schema update for the first for a large connection.
`change.schedule_job` will be `None` but frontend still needs to show in advance at what time this will be scheduled if they accept schema changes

Reply:
ah all right then